server/src/main/java/org/opensearch/index/store/Store.java

```diff
@@ -1102,33 +1102,13 @@ public Map<String, StoreFileMetadata> asMap() {
         private static final String LIV_FILE_EXTENSION = "liv"; // lucene 5 delete file
         private static final String SEGMENT_INFO_EXTENSION = "si";
 
-        /**
-         * Helper method used to group store files according to segment and commit.
-         *
-         * @see MetadataSnapshot#recoveryDiff(MetadataSnapshot)
-         * @see MetadataSnapshot#segmentReplicationDiff(MetadataSnapshot)
-         */
-        private Iterable<List<StoreFileMetadata>> getGroupedFilesIterable() {
-            final Map<String, List<StoreFileMetadata>> perSegment = new HashMap<>();
-            final List<StoreFileMetadata> perCommitStoreFiles = new ArrayList<>();
-            for (StoreFileMetadata meta : this) {
-                final String segmentId = IndexFileNames.parseSegmentName(meta.name());
-                final String extension = IndexFileNames.getExtension(meta.name());
-                if (IndexFileNames.SEGMENTS.equals(segmentId)
-                    || DEL_FILE_EXTENSION.equals(extension)
-                    || LIV_FILE_EXTENSION.equals(extension)) {
-                    // only treat del files as per-commit files fnm files are generational but only for upgradable DV
-                    perCommitStoreFiles.add(meta);
-                } else {
-                    perSegment.computeIfAbsent(segmentId, k -> new ArrayList<>()).add(meta);
-                }
-            }
-            return Iterables.concat(perSegment.values(), Collections.singleton(perCommitStoreFiles));
-        }
-
         /**
          * Returns a diff between the two snapshots that can be used for recovery. The given snapshot is treated as the
-         * recovery target and this snapshot as the source. The returned diff will hold a list of files that are:
+         * recovery target and this snapshot as the source. This method is also used for segment replication where
+         * per-segment missing files are not considered inconsistent due to replication lag between primary and
```

Comment:
I'm not sure the replication lag is the reason here for this inconsistency.

```java
This method is also used for segment replication where per-segment missing files does not result in the entire segment group being marked as different.
```

Reply:
Updated